src/detectors/signal_analyzers/assurance.hpp

```diff
@@ -132,12 +132,7 @@ class AssuranceDetector : public BaseDetector {
   void shiftBasePoints();
 
   /**
-   * Contention Factory.
-   */
-  Detection createContention(double_t severity);
-
-  /**
-   * It is possible to dynamically change detector configuration.
+   * It is possible to dynamically change analyzer configuration.
    */
   void recalculateParams();
```

Comment:
RecalculateParams should be protected and it should be invoked whenever it has to be invoked (eg. some important param has changed and Filter needs to recalculate things).
Consider adding public accesors for your (then ```protected```) members and invoke this function wherever it has to be.